e2e-polybft/cardanofw/test-apex-system.go

```diff
@@ -501,6 +517,58 @@ func (a *ApexSystem) GetChainMust(t *testing.T, chainID string) ITestApexChain {
 	return chain
 }
 
+func WaitForBatchState(
+	ctx context.Context, requestURL string, apiKey string, breakIfFailed bool, failAtLeastOnce bool, batchState string,
+) (int, bool) {
+	var (
+		prevStatus           string
+		currentStatus        string
+		failedToExecuteCount int
+		timeout              bool
+	)
+
+	timeoutTimer := time.NewTimer(time.Second * 300)
```

Comment:
why dont just use `context.WithTimeout` ? we do not need both cases (`<-timeoutTimer.C` and `<-ctx.Done`)